README.dev.md

```diff
@@ -6,11 +6,612 @@ You will need a machine with at least 2GB of RAM available (see
 https://github.com/zulip/zulip/issues/32 for a plan for how to
 dramatically reduce this requirement).
 
-Start by cloning this repository: `git clone https://github.com/zulip/zulip.git`
+Contents:
+
+* [First-time contributors start here](#first-time-contributors-start-here)
+* [Using Vagrant](#using-vagrant)
+* [Using provision.py without Vagrant](#using-provisionpy-without-vagrant)
+* [By hand](#by-hand)
+* [Using Docker (experimental)](#using-docker-experimental)
+
+## First-time contributors start here
+
+This section guides first-time contributors through installing the Zulip dev
+environment on Windows 10, OS X El Capitan, Ubuntu 14.04, and Ubuntu 16.04. 
+
+Those who have installed Zulip before or are experienced at using Vagrant may
+wish to skip ahead to [Using Vagrant](#using-vagrant), [Using
+Docker](#using-docker-experimental), or read about installing [By
+hand](#by-hand).
```

Comment:
I think that these introductory sections need to be more clear that you can use Zulip in development on a wider range of Linux/BSD/etc., and it's documented in the "by hand" installation sections.